### DISCUSSION TITLE
How to use variables in `.md` files

### DISCUSSION THREAD

**Discussion by adebiyial** [markdoc/markdoc#266] [Q&A] [answered]
I've read the documentation on [Variables](https://markdoc.dev/docs/variables) but I still don't understand how to use it.

First, where does the [Config](https://markdoc.dev/docs/syntax#config) file goes?
Second, do I need to `.parse` and `.transform` in the `.md` file to use the variables?

**Comment by mfix-stripe** [comment, marked as answer]
Hey @adebiyial 👋 

Once you have the contents of you `.md` file, that's when you call `parse` and then `transform`. 

Here is an end to end example:
```js
const Markdoc = require('@markdoc/markdoc')

const source = `

{% $myVariable %}

`

const ast = Markdoc.parse(source)

const content = Markdoc.transform(ast, {
  variables: {
    myVariable: 'Content goes here'
  }
})

const html = Markdoc.renderers.html(content)

```

**Reply by adebiyial** [reply]
Thank you @mfix-stripe. I was thinking I could use the variables in the `.md` files directly?

**Reply by mfix-stripe** [reply]
@adebiyial not sure how you imagine doing that 🙂 The `.md` file is a static document—it's not "code" per se. This is an intentional design constraint of Markdoc, keeping your content separate from your code. All the code logic is done in JavaScript. Markdoc runs this logic and takes your static `.md` document and transforms it into React or HTML, depending which renderer you choose.

**Reply by adebiyial** [reply]
Hmm. I get that now. My use-case sort of led me down that path. In our docs, we have variables in `.md` files and use Webpack to replace the variables on the fly. But I understand better now, thank you!

**Reply by mfix-stripe** [reply]
@adebiyial you can use Markdoc to do the variable replacement! Every time you call `.transform`, a new output will get created with the new variables you pass to the second parameter of `.transform` 🙂

**Reply by adebiyial** [reply]
Got it. Thank you for the explanation. I appreciate it.
